src/stores/user.ts

```diff
@@ -51,26 +51,51 @@ export const useUserStore = defineStore("userStore", {
       }
     },
     async login(credentials: LoginCredentials) {
-      if (this.isLoggedIn) await logout();
-      const response = await login(credentials);
-      const authHeader = response.headers.authorization;
-      const authToken = authHeader.slice("Bearer ".length);
-      setAuthHeader(authHeader);
-      this.setUserData({ authToken, user: response.data });
-      router.push({ name: "HomePage" });
+      try {
+        if (this.isLoggedIn) await logout();
```

Comment:
This piece could be extracted probably to another method since it's being used in other places too.